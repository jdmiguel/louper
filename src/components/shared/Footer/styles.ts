import { styled } from '@mui/material/styles';
import Stack from '@mui/material/Stack';

export const StyledRoot = styled('footer')(({ theme }) => ({
  alignItems: 'center',
  bottom: 0,
  color: theme.palette.secondary.light,
  display: 'none',
  fontSize: '0.9rem',
  padding: '12px 24px',
  position: 'absolute',
  width: '100%',
  '@media (min-width: 500px)': {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  span: {
    marginRight: 5,
  },
}));

export const StyledContent = styled(Stack)({
  marginRight: 0.6,
});
