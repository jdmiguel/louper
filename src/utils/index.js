export const errorLiterals = {
  maximumRequest:
    'You have excedeed the maximum allowed request. Please, wait for a while',
  unavailableUser: 'Please, choose an available user',
};

export const externalLink = (url) => window.open(url, '_blank');
