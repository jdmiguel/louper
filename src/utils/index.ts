import { ResponseError } from './types';

enum ErrorMessage {
  MAX = 'You have excedeed the maximum allowed request. Please, wait for a while',
  NO_USER = 'Please, choose an available user',
  GENERIC = 'Sorry! there was an error on the server side.',
}

export const getErrorMessage = (error: ResponseError) => {
  if (error.status === 403) {
    return ErrorMessage.MAX;
  }

  if (error.status === 404) {
    return ErrorMessage.NO_USER;
  }

  return ErrorMessage.GENERIC;
};

export const debounce = (fn: any, delay: number) => {
  let timer: ReturnType<typeof setTimeout>;

  return (...args: any) => {
    if (timer) clearTimeout(timer);
    timer = setTimeout(() => {
      fn(...args);
    }, delay);
  };
};
