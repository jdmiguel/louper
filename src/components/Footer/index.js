import React from 'react';

/* styles */
import './styles.css';

const Footer = () => (
  <div className="footer">
    <h3>
      Created by{' '}
      <a
        href="https://jdmiguel.netlify.app/"
        target="_blank"
        rel="noopener noreferrer"
        aria-label="View source on GitHub"
      >
        jdmiguel
      </a>
    </h3>
  </div>
);

export default Footer;
