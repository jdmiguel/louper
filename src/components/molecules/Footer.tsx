/* material-ui */
import { styled } from '@mui/material/styles';
import Typography from '@mui/material/Typography';

/* atoms */
import Link from '../atoms/Link';

const Root = styled('footer')({
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  opacity: 1,
  padding: '12px 24px',
  position: 'absolute',
  bottom: 0,
  visibility: 'visible',
  width: '100%',
});

const Footer = () => {
  return (
    <Root>
      <Typography variant="body1" sx={{ marginRight: 0.7 }}>
        COPYRIGHT @2021 | Created by{' '}
      </Typography>
      <Link
        url="https://jdmiguel.netlify.app/"
        ariaLabel="View profile of jdmiguel on GitHub"
        content="jdmiguel"
      />
    </Root>
  );
};

export default Footer;
