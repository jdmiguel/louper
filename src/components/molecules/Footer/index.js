/* material-ui */
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';

const useStyles = makeStyles((theme) => ({
  wrapper: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    padding: '0 16px 12px',
    width: '100%',
    height: '5vh',
    '@media (min-width: 768px)': {
      position: 'absolute',
      left: 0,
      bottom: 0,
      zIndex: 2,
      height: 'auto',
    },
  },
  link: {
    fontWeight: 700,
  },
}));

const Footer = () => {
  const classes = useStyles();

  return (
    <footer className={classes.wrapper}>
      <Typography variant="body2">
        COPYRIGHT @2021 | Created by{' '}
        <Link
          className={classes.link}
          href="https://jdmiguel.netlify.app/"
          target="_blank"
          rel="noopener noreferrer"
          aria-label="View source on GitHub"
        >
          jdmiguel
        </Link>
      </Typography>
    </footer>
  );
};

export default Footer;
