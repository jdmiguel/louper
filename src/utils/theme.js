/* material-ui */
import { keyframes } from '@emotion/react';
import { createTheme } from '@mui/material/styles';

export const theme = createTheme({
  overrides: {
    MuiCssBaseline: {
      '@global': {
        html: {
          overflowX: 'hidden',
          background: '#E0E6EF',
        },
        body: {
          margin: 0,
          padding: 0,
        },
        button: {
          boxShadow: 'initial !important',
        },
      },
    },
  },
  typography: {
    fontFamily: ['Comfortaa', 'Arial', 'sans-serif'].join(','),
    htmlFontSize: 12,
    h1: {
      color: '#353D40',
      fontSize: '4.8rem',
      fontWeight: 700,
      letterSpacing: '-0.25rem',
      lineHeight: 0,
      '@media (min-width: 769px)': {
        fontSize: '6rem',
      },
    },
    h2: {
      color: '#353D40',
      fontSize: '3.5rem',
      fontWeight: 300,
      letterSpacing: '-0.1rem',
      lineHeight: 0,
      '@media (min-width: 769px)': {
        fontSize: '4.2rem',
      },
    },
    h3: {
      color: '#353D40',
      fontSize: '1.3rem',
      fontWeight: 700,
      '@media (min-width: 769px)': {
        fontSize: '1.5rem',
      },
    },
    h4: {
      color: '#353D40',
      fontSize: '1.1rem',
      fontWeight: 500,
      '@media (min-width: 769px)': {
        fontSize: '1.2rem',
      },
    },
    h5: {
      color: '#6b77a1',
      fontSize: '0.9rem',
      fontWeight: 700,
      textTransform: 'uppercase',
    },
    h6: {
      color: '#757d81',
      fontSize: '0.8rem',
      fontWeight: 500,
      textTransform: 'uppercase',
    },
    button: {
      fontSize: '0.9rem',
    },
    body1: {
      color: '#353D40',
      fontSize: '0.9rem',
      lineHeight: '1.1rem',
    },
    body2: {
      color: '#E1D7EC',
      fontSize: '0.8rem',
      lineHeight: '1rem',
    },
    subtitle2: {
      color: '#353D40',
      fontSize: '0.8rem',
      fontWeight: 500,
      textTransform: 'uppercase',
    },
  },
  palette: {
    background: {
      default: '#E0E6EF',
    },
    primary: {
      main: '#553285',
      light: '#7B52AB',
      dark: '#36175E',
      contrastText: '#E1D7EC',
    },
    secondary: {
      main: '#6b77a1',
      light: '#c7cde1',
      dark: '#565e7a',
      contrastText: '#bac1d9',
    },
    neutral: {
      main: '#353D40',
    },
  },
  animation: {
    greet: keyframes`
      0%,100% { transform: rotate(0) }
      20%,60% { transform: rotate(-25deg) }
      40%,80% { transform: rotate(10deg) }
    `,
  },
});
