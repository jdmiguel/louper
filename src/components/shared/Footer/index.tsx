import Stack from '@mui/material/Stack';
import { FOOTER_CONTENT, FOOTER_LINKS } from '@/utils/literals';
import Link from '../Link';
import { StyledRoot, StyledContent } from './styles';

const [portfolioLink, octoverseLink] = FOOTER_LINKS;

const Footer = () => (
  <StyledRoot data-testid="footer">
    <StyledContent direction="row">
      <span>{FOOTER_CONTENT.creation}</span>
      <Link
        url={portfolioLink.url}
        ariaLabel={portfolioLink.ariaLabel}
        content={portfolioLink.content}
      />
    </StyledContent>
    <span>|</span>
    <Stack direction="row">
      <span>{FOOTER_CONTENT.stats}</span>
      <Link
        url={octoverseLink.url}
        ariaLabel={octoverseLink.ariaLabel}
        content={octoverseLink.content}
      />
    </Stack>
  </StyledRoot>
);

export default Footer;
