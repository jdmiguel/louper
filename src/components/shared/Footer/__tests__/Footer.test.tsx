import { render } from '@testing-library/react';
import { renderWithTheme } from '../../../../utils/theme';
import Footer from '..';

describe('<Footer />', () => {
  it('displays the content properly', () => {
    const { container } = render(renderWithTheme(<Footer />));

    expect(container).toMatchSnapshot();
  });
});
