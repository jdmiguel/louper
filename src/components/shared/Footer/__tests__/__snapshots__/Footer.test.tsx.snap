// Vitest Snapshot v1

exports[`<Footer /> > displays the content properly 1`] = `
<div>
  <footer
    class="css-16wb799"
    data-testid="footer"
  >
    <div
      class="MuiStack-root css-1oy7hqd-MuiStack-root"
    >
      <span>
        @2023 | created by
      </span>
      <a
        aria-label="View profile of jdmiguel on GitHub"
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1o7qz1o-MuiTypography-root-MuiLink-root"
        href="https://jdmiguel.com/"
        rel="noopener noreferrer"
      >
        jdmiguel
      </a>
    </div>
    <span>
      |
    </span>
    <div
      class="MuiStack-root css-m69qwo-MuiStack-root"
    >
      <span>
        github stats from
      </span>
      <a
        aria-label="View github users statistics on octoverse"
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1o7qz1o-MuiTypography-root-MuiLink-root"
        href="https://octoverse.github.com/#future"
        rel="noopener noreferrer"
      >
        octoverse
      </a>
    </div>
  </footer>
</div>
`;
